neopixel.py

```diff
@@ -112,9 +112,11 @@ def __init__(self, pin, n, *, bpp=3, brightness=1.0, auto_write=True, pixel_orde
                     order[pixel_order] = order_chars[char_no]
                 pixel_order = ''.join(order)
 
-        super().__init__(n, bytearray(self.n * bpp),
+        self._buf = bytearray(self.n * bpp)
```

Comment:
ideally we would not allocate two buffers if they are not needed - the original code only alloc'd a new buffer when brightness was < 1